### PR TITLE
Add CRD validation for fqdn rules

### PR DESCRIPTION
**Pull request by nebril**
Fixes: #6276

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6414)

### CODE REVIEW

File: pkg/k8s/apis/cilium.io/v2/register.go [hunk 1]
```diff
@@ -548,9 +548,37 @@ var (
 					"AWS": AWSGroup,
 				},
 			},
+			"toFQDNs": {
```
Comment (aanm) [inline pkg/k8s/apis/cilium.io/v2/register.go:555]: But the CRD validation version, it should be on this file

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/apis/cilium.io/v2/register.go (hunk 1 above)
Comment (nebril) [reply pkg/k8s/apis/cilium.io/v2/register.go:555]: You mean `CustomResourceDefinitionSchemaVersion` variable, right? True, forgot about it.

**Comment by nebril** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: pkg/k8s/apis/cilium.io/v2/register.go [hunk 2]
```diff
@@ -548,9 +548,37 @@ var (
...
+		Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+			"matchName":    MatchFQDNName,
+			"matchPattern": MatchFQDNPattern,
+		},
+	}
+
+	MatchFQDNName = apiextensionsv1beta1.JSONSchemaProps{
+		Description: `MatchName matches fqdn name`,
+		Type:        "string",
+		Pattern:     `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`,
```
Comment (raybejjani) [inline pkg/k8s/apis/cilium.io/v2/register.go]: I think both of these need a `\.?` at the end, they don't allow `foo.com.` type names.

**Review by raybejjani — CHANGES_REQUESTED** [review]

File: pkg/k8s/apis/cilium.io/v2/register.go (hunk 2 above)
Comment (aanm) [reply pkg/k8s/apis/cilium.io/v2/register.go]: @nebril As this is a regex can you create unit tests for the regex itself and make the regex as a constant in this file?

Comment (nebril) [reply pkg/k8s/apis/cilium.io/v2/register.go]: done

**Comment by nebril** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

**Pull request merged**